node/src/components/small_network/message.rs

```diff
@@ -2,16 +2,25 @@ use std::fmt::{self, Debug, Display, Formatter};
 
 use serde::{Deserialize, Serialize};
 
+use crate::crypto::hash::Digest;
+
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub enum Message<P> {
-    Handshake { network_name: String },
+    Handshake {
+        genesis_config_hash: Digest,
```

Comment:
Should it be more generally – `chainspec_config_hash`? I guess that after an upgrade it's going to be a hash of the upgrade config, not the old Genesis one.

Reply:
Ah, but the nodes running software with upgrade will never send us the config hash but the network name.